src/components/CytoscapeGraph/CytoscapeTrace.ts

```diff
@@ -0,0 +1,95 @@
+import * as Cy from 'cytoscape';
+import { CyNode } from './CytoscapeGraphUtils';
+import { JaegerTrace, Span } from 'types/JaegerInfo';
+import {
+  getWorkloadFromSpan,
+  searchParentWorkload as getParentWorkloadFromSpan
+} from 'components/JaegerIntegration/JaegerHelper';
+import { NodeType } from 'types/Graph';
+
+export const showTrace = (cy: Cy.Core, trace: JaegerTrace) => {
+  if (!cy) {
+    return;
+  }
+
+  cy.startBatch();
+  hideTrace(cy);
+
+  trace.spans.forEach(span => {
+    const split = span.process.serviceName.split('.');
+    const service = split[0];
+    // TODO: add ns when missing
```

Comment:
I think we can remove this "todo"; if namespace isn't part of trace's service name, we have no reliable way to guess it, so I think how you did is fine: select node only from service.
This should never happen with envoy-generated traces, but it may happen with user-defined traces

Reply:
+1, I left it there but I didn't know what else to do to be able to guess a valid namespace.